packages/eventcatalog-plugin-generator-asyncapi/src/__tests__/plugin.spec.ts

```diff
@@ -243,6 +246,56 @@ describe('eventcatalog-plugin-generator-asyncapi', () => {
               producers:
                   - 'Account Service'
               consumers: []
+              externalLinks: []
+            ---
+
+            <Mermaid />
+            <Schema />
+            `);
+        });
+      });
+
+      describe('includeLinkToAsyncAPIDoc', () => {
+        it('when includeLinkToAsyncAPIDoc is set, an external link will be added in the event', async () => {
+          const options: AsyncAPIPluginOptions = {
+            pathToSpec: path.join(__dirname, './assets/valid-asyncapi.yml'),
+            externalAsyncAPIUrl: 'https://eventcatalog.dev/events',
+          };
+
+          const oldEvent = {
+            name: 'UserSignedUp',
+            version: '0.0.1',
+            summary: 'Old example of an event that should be versioned',
+            producers: ['Service A'],
+            consumers: ['Service B'],
+            owners: ['dBoyne'],
+          };
+
+          const { writeEventToCatalog } = utils({ catalogDirectory: process.env.PROJECT_DIR });
+          const { path: eventPath } = await writeEventToCatalog(oldEvent, {
+            schema: { extension: 'json', fileContent: 'hello' },
+          });
+
+          // run plugin
+          await plugin(pluginContext, options);
+
+          const { getEventFromCatalog } = utils({ catalogDirectory: process.env.PROJECT_DIR });
+          const { raw: eventFile } = getEventFromCatalog('UserSignedUp');
+
+          // Check the file has been created
+          expect(fs.existsSync(path.join(eventPath, 'index.md'))).toEqual(true);
+          expect(fs.existsSync(path.join(eventPath, 'schema.json'))).toEqual(true);
+
+          expect(eventFile).toMatchMarkdown(`
+            ---
+              name: UserSignedUp
+              summary: null
+              version: 1.0.0
+              producers:
+                  - 'Account Service'
+              consumers: []
+              externalLinks:
+                  - {label: 'View event in AsyncAPI', url: 'https://eventcatalog.dev/events#message-UserSignedUp'}
```

Comment:
@boyney123 Are you OK with this syntax? 

```yaml
externalLinks:
                  - {label: 'View event in AsyncAPI', url: 'https://eventcatalog.dev/events#message-UserSignedUp'}
                  ```
                  
 This is how the writeToCatalog injects it. It works since it is a valid syntax, but it looks strange in a YAML file.

Reply:
Yeah think it's OK for now, I was shocked by the format when I saw it first last week, but as you said seems to be valid YML. The lib we use seems to format arrays like this 🤷‍♂️, guess if its a major issue we can fix in the future 👍